samples/cpp/dis_opticalflow.cpp

```diff
@@ -2,30 +2,24 @@
 #include "opencv2/core/utility.hpp"
 #include "opencv2/highgui.hpp"
 #include "opencv2/imgproc.hpp"
+#include "opencv2/videoio.hpp"
 #include "opencv2/video.hpp"
 
 using namespace std;
 using namespace cv;
 
-static void help()
-{
-    printf("Usage: dis_optflow <video_file>\n");
-}
-
 int main(int argc, char **argv)
 {
-    VideoCapture cap;
+    CommandLineParser parser(argc, argv, "{ @video  | vtest.avi  | use video as input }");
+    string filename = samples::findFile(parser.get<string>("@video"));
```

Comment:
Please use `samples::findFileOrKeep()` for VideoCapture inputs (to support non-files too: URLs for IP cameras or GStreamer pipelines)